pkg/apis/apps/v1beta2/defaults_test.go

```diff
@@ -416,13 +406,6 @@ func TestSetDefaultReplicaSet(t *testing.T) {
 			t.Errorf("unexpected object: %v", rs2)
 			t.FailNow()
 		}
-		if test.expectSelector != reflect.DeepEqual(rs2.Spec.Selector.MatchLabels, rs2.Spec.Template.Labels) {
```

Comment:
Do not default the selector to anything. We want to eliminate setting the Selector, or any field for that matter, based on another field of the same object. Doing so has caused us pain wrt apply and patch.